### DISCUSSION TITLE
Transaction commit hooks

### DISCUSSION THREAD

**Discussion by shreyasb22** [JanusGraph/janusgraph#3790] [Q&A]
**Describe the feature:**
I have noticed that in the `JanusGraphTransaction` interface,  there are no functional implementations for beforeCommit and afterCommit hooks. The same goes for beforeRollback and afterRollback as well, both of which are available in Neo4j. I wanted to build upon this interface and implement an interceptor for my custom logic. Are there any alternatives for this? Or any plans to include the functionality in subsequent versions of Janusgraph?

**Comment by porunov** [comment]
Are you planning to implement a blocking interceptor or a non-blocking notification system? In case you need to block the transaction until your logic approves it then there is no such feature yet. I don’t think it’s hard to implement one, but I’m not aware of anyone working on this feature.
In case you need some notification system then you can refer to WAL to notify your system whenever any (or specific) transaction commit begins and use Transaction Log to notify whenever any (or specific) transactions commit ends. 
See documentation here about Transaction Log: https://docs.janusgraph.org/advanced-topics/transaction-log/

**Reply by shreyasb22** [reply]
I want to implement a blocking interceptor and add some checks before and after the commit function executes. I was able to implement that logic on my own after creating a simple POC for the same. Thank you for the reply!

Looking forward to the possibility of having this functionality provided out-of-the-box in the future. It would be beneficial to have built-in support for interceptors and customisable transaction behaviour in JanusGraph.
